### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by ivan-kadykalo**
[DEMO LINK](https://ivan-kadykalo.github.io/react_dynamic-list-of-todos/)

### CODE REVIEW

**Review by OksanaTyshchenko — APPROVED** [review]
As for me looks good! Great job!

File: src/App.tsx [hunk 1]
```diff
@@ -1,28 +1,61 @@
...
 import { TodoList } from './components/TodoList';
 import { CurrentUser } from './components/CurrentUser';
+import { API_URL, request } from './components/API/api';
 
 const App: React.FC = () => {
-  const [
-    selectedUserId,
-    // setSelectedUserId,
-  ] = useState(0);
+  const [selectedUserId, setSelectedUserId] = useState(0);
```
Comment (AlinaMatuschak) [inline src/App.tsx:9]:
```suggestion
  const [selectedUserId, setSelectedUserId] = useState(null);
```

File: src/App.tsx [hunk 2]
```diff
@@ -1,28 +1,61 @@
...
+  const [todos, setTodos] = useState([]);
+  const [downloadError, setDownloadError] = useState('');
+
+  useEffect(() => {
+    const getDataFromServer = async () => {
+      try {
+        const dataFromServer = await request(API_URL, '');
```
Comment (AlinaMatuschak) [inline src/App.tsx:16]: create all the API calls (load user and todos) on api file (don't need export API_URL from api file)

File: src/components/CurrentUser/CurrentUser.tsx [hunk 3]
```diff
@@ -1,12 +1,75 @@
...
+  useState,
+} from 'react';
+import { request, USERS_URL } from '../API/api';
 import './CurrentUser.scss';
 
-export const CurrentUser: React.FC = () => (
-  <div className="CurrentUser">
-    <h2 className="CurrentUser__title"><span>Selected user: 2</span></h2>
+enum Error {
+  Initial = '',
```
Comment (AlinaMatuschak) [inline src/components/CurrentUser/CurrentUser.tsx:11]:
```suggestion
  Initial = null,
```

File: src/components/TodoList/TodoList.tsx [hunk 4]
```diff
@@ -1,44 +1,123 @@
...
+export const TodoList: FC<Props> = ({
+  todos,
+  selectedUserId,
+  selectNewUser,
+}) => {
+  const [filterQuery, setFilterQuery] = useState('');
+  const [status, setStatus] = useState(Status.All);
+  const [shuffle, setShuffle] = useState(false);
+
+  const filteredStatus = () => {
```
Comment (AlinaMatuschak) [inline src/components/TodoList/TodoList.tsx:31]:
```suggestion
  const filteredTodosByStatus = () => {
```

File: src/components/TodoList/TodoList.tsx [hunk 5]
```diff
@@ -1,44 +1,123 @@
...
+    setStatus(event.target.value as Status);
+  }, []);
+
+  const changeFilter = useCallback((event: ChangeEvent<HTMLInputElement>) => {
+    setFilterQuery(event.target.value);
+  }, []);
+
+  const changeShuffle = useCallback(() => {
+    setShuffle(prev => !prev);
+  }, []);
```
Comment (AlinaMatuschak) [inline src/components/TodoList/TodoList.tsx:64]: you should on every click create random orders for todos.
Now every second click returns the orders to default

**Review by AlinaMatuschak — CHANGES_REQUESTED** [review]
You don't need to render the user button if you don’t have user id

Screenshot 2022-05-23 at 12 31 52

File: src/components/CurrentUser/CurrentUser.tsx [hunk 6]
```diff
@@ -1,12 +1,70 @@
...
+        setUserError('Can\'t download user');
+      }
+    };
+
+    getUser();
+  }, [userId]);
+
+  const clearUser = useCallback(() => {
+    selectNewUser(0);
+  }, []);
```
Comment (AlinaMatuschak) [inline src/components/CurrentUser/CurrentUser.tsx:40]: but default value for user id is null

**Pull request closed without merging**